lib/util.c

```diff
@@ -2187,3 +2188,61 @@ EXPORTED char *modseqtoa(modseq_t modseq)
     buf_printf(&buf, MODSEQ_FMT, modseq);
     return buf_release(&buf);
 }
+
+EXPORTED void buf_append_sgr(struct buf *dst, ...)
+{
+    va_list ap;
+    int n, sep = 0;
+
+    buf_appendcstr(dst, "\033[");
+    va_start(ap, dst);
+    while ((n = va_arg(ap, int)) >= 0) {
+        if (sep) buf_putc(dst, sep);
+        buf_printf(dst, "%d", n);
+        sep = ';';
+    }
+    buf_putc(dst, 'm');
+}
+
+EXPORTED void buf_append_kv(struct buf *dst, int sep, int want_color,
```

Comment:
Ooooh I don't like that!  Maybe it saves a register, but it's a lot harder to read and reason about if you're not already elbows deep in it.

I'd been thinking more about `want_color`, because maybe you want a deterministic colour based on the hashed string contents (which is what you get with this flag at present), but maybe you want to just give it a specific colour index (0-255, corresponding with the xterm-256 palette).  And I was thinking of making _that_ option a squashed "either it's a value, or it's a flag" thing, but kept not doing it because it bothered me.

There's room for this kv API to improve as it gets used for real... I've written this sort of thing in different ways for different projects a bunch of times over the years, and each time I do it it gets a little bit better, but I'm never quite convinced that it's the last time I'll rewrite and improve it, cause it's never quite perfect...